tests/meta/test_type_traits.cpp

```diff
@@ -0,0 +1,34 @@
+#include <tuple>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "kokkos-utils/meta/type_traits.hpp"
+
+/**
+ * @file
+ *
+ * @addtogroup unittests
+ *
+ * **Type traits**
+ *
+ * This group of tests check the behavior of our utilities extending the standard @c type_traits.
+ */
+
+namespace Kokkos::utils::tests::meta
+{
+
+//! @test Check @ref Kokkos::utils::meta::is_instance_of_v.
+TEST(Traits, is_instance_of)
+{
+    using Kokkos::utils::meta::is_instance_of_v;
+
+    static_assert(  is_instance_of_v<std::vector<int   >, std::vector>);
+    static_assert(  is_instance_of_v<std::vector<double>, std::vector>);
+    static_assert(! is_instance_of_v<std::tuple <double>, std::vector>);
+
+    static_assert(  is_instance_of_v<std::tuple<int, double>, std::tuple>);
+    static_assert(! is_instance_of_v<std::tuple<int, double>, std::vector>);
```

Comment:
Could be good to add the case of `std::array` to see whether this works with template parameters that are not types.

```
static_assert(is_instance_of_v<std::array<int, 3>, std::array>);
```
(actually I'm not sure if we should expect it to work)

Reply:
This won't work with `std::array` because the size is a non-type template argument. We would need to provide specializations for `std::array`, but we agreed not to do it, because we might never need it.